service/internal/nodecount/node_count.go

```diff
@@ -132,14 +138,15 @@ func (nc *NodeCount) cachedNodes(ctx context.Context, cr metav1.Object) (corev1.
 	return nodes, nil
 }
 
-func (nc *NodeCount) lookupNodes(ctx context.Context) (corev1.NodeList, error) {
-	// TODO we need to get rid off the controllercontext.Context but for now it should be fine.
-	cc, err := controllercontext.FromContext(ctx)
-	if err != nil {
+func (nc *NodeCount) lookupNodes(ctx context.Context, cr metav1.Object) (corev1.NodeList, error) {
+	client, err := nc.tenantClient.K8sClient(ctx, cr)
+	if tenantclient.IsNotAvailable(err) {
+		return corev1.NodeList{}, microerror.Mask(tenantClusterNotInitializedError)
```

Comment:
I guess when this error is new there needs to be handling for it now. Is that right?

Reply:
This is the same error as before. node count will be called only by those two resources (machinedeploymentstatus and controlplanestatus). They are checking if the tentantcluster is not initialized that's why I'm returning this error in case of tenantclient is not available.

https://github.com/giantswarm/cluster-operator/blob/master/service/controller/resource/controlplanestatus/resource.go#L73
https://github.com/giantswarm/cluster-operator/blob/master/service/controller/resource/machinedeploymentstatus/resource.go#L74

Reply:
I see. That should be good then. Just noticed that it is `IsTenantClusterInitialized` which means the `Not` is missing.